fbw-a32nx/src/wasm/systems/a320_systems/src/hydraulic/sfcc/slats_channel.rs

```diff
@@ -29,14 +33,31 @@ impl SlatsChannel {
             slats_fppu_angle_id: context.get_identifier("SLATS_FPPU_ANGLE".to_owned()),
             slat_actual_position_word_id: context
                 .get_identifier(format!("SFCC_{num}_SLAT_ACTUAL_POSITION_WORD")),
+            sap_ids: [1, 2, 3, 4, 5, 6, 7]
+                .map(|id| context.get_identifier(format!("SFCC_{num}_SAP_{id}"))),
 
             slats_demanded_angle: Angle::ZERO,
             slats_feedback_angle: Angle::ZERO,
 
             csu_monitor: CSUMonitor::new(context),
+
+            // Set `sap` to false to match power-off state
+            sap: [false; 7],
         }
     }
 
+    fn sap_update(&mut self) {
+        let fppu_angle = self.slats_feedback_angle.get::<degree>();
+
+        self.sap[0] = fppu_angle > -5.0 && fppu_angle < 6.2;
+        self.sap[1] = fppu_angle > 198.0 && fppu_angle < 337.0;
+        self.sap[2] = fppu_angle > 210.4 && fppu_angle < 337.0;
+        self.sap[3] = fppu_angle > -5.0 && fppu_angle < 6.2;
+        self.sap[4] = fppu_angle > 210.4 && fppu_angle < 337.0;
+        self.sap[5] = fppu_angle > 198.0 && fppu_angle < 337.0;
```

Comment:
These look very similar to the conditions for sap 0-2 only that 4 and 5 are in a different order. Is this intended?

Reply:
Yes they are. The reason being that it’s a discrete signal directly connected to the receiver. They still need to be used in the a32nx